### PR TITLE
warn on py3.4 and py2.7 eol dates for cfn-lint

### PR DESCRIPTION
**Pull request by kddejong**
*Issue #, if available:*

*Description of changes:*
- warn on py3.4 eol date
- warn on py2.7 eol date

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/cfnlint/__main__.py [hunk 1]
```diff
@@ -12,6 +13,13 @@
 
 def main():
     """Main function"""
+    if sys.version_info[:2] == (3, 4):
+        warnings.warn('Python 3.4 will reach the end of its life on March 18th, 2019. '
```
Comment (PatMyron) [inline src/cfnlint/__main__.py]:
```suggestion
        warnings.warn('Python 3.4 has reached end of life. '
```

File: src/cfnlint/__main__.py [hunk 2]
```diff
@@ -12,6 +13,13 @@
...
+                      'cfn-lint will end support for python 3.4 on July 1st, 2020.', Warning, stacklevel=3)
+    elif sys.version_info[:2] == (2, 7):
+        warnings.warn('Python 2.7 will reach the end of its life on January 1st, 2020. '
```
Comment (PatMyron) [inline src/cfnlint/__main__.py]:
```suggestion
        warnings.warn('Python 2.7 has reached end of life. '
```

**Comment by PatMyron** [comment]
@kddejong as long as this doesn't change the exit code

**Comment by kddejong** [comment]
Should be good there
```
➜  cfn-lint test/fixtures/templates/bad/resources/dynamodb/undefined_attribute_definition.yaml                                                                               
sys:1: Warning: Python 2.7 will reach the end of its life on January 1st, 2020. cfn-lint will end support for python 2.7 on December 31st, 2020.
E3039 The set of Attributes in AttributeDefinitions: ['Album'] and KeySchemas: ['Album', 'Artist'] must match at Resources/dynamoDBTable/Properties
test/fixtures/templates/bad/resources/dynamodb/undefined_attribute_definition.yaml:6:5

➜  echo $?                                                                                                                                                                  
2
➜  cfn-lint test/fixtures/templates/public/rds-cluster.yaml                                                                                                                  
sys:1: Warning: Python 2.7 will reach the end of its life on January 1st, 2020. cfn-lint will end support for python 2.7 on December 31st, 2020.
➜  echo $?                                                                                                                                                                  
0
```

**Comment by kddejong** [comment]
This will show up in visual studio though.
```
[cfn-lint] sys:1: Warning: Python 2.7 will reach the end of its life on January 1st, 2020. cfn-lint will end support for python 2.7 on December 31st, 2020.
```

**Comment by gliptak** [comment]
https://github.com/aws-cloudformation/cfn-python-lint/issues/1291

**Pull request merged**